parsing/parser.mly

```diff
@@ -1816,6 +1810,8 @@ simple_pattern_not_ident:
       { $1 }
   | mod_longident DOT simple_delimited_pattern
       { mkpat @@ Ppat_open(mkrhs $1 1, $3) }
+  | mod_longident DOT LBRACKET RBRACKET
+    { mkpat @@ Ppat_construct ( mkrhs (Lident "[]") 1, None) }
```

Comment:
That's right, you should wrap a `Ppat_open` around the `Ppat_construct`.

@Drup the syntax was introduced in trunk some time ago.